Scalar.Common/ScalarPlatform.cs

```diff
@@ -131,6 +131,28 @@ public bool TryGetNormalizedPathRoot(string path, out string pathRoot, out strin
             return true;
         }
 
+        public bool IsFileSystemCaseSensitivitySupported(string path, out string errorMessage)
+        {
+            Exception ex = null;
+            bool caseSensitive = IsDirectoryCaseSensitive(path, out ex);
+
+            if (ex != null)
+            {
+                errorMessage = $"Exception when performing {nameof(this.IsFileSystemCaseSensitivitySupported)}: {ex.ToString()}";
+                return false;
+            }
+
+            bool caseSensitiveFileSystem = this.Constants.CaseSensitiveFileSystem;
+            if (caseSensitive != caseSensitiveFileSystem)
+            {
+                errorMessage = $"Scalar does not support case {(caseSensitiveFileSystem ? "in" : "")}sensitive filesystems on {this.Name}";
```

Comment:
This could use an update in the `docs/faq.md` possibly. Perhaps create an issue for supporting non-standard case-sensitivities? We could theoretically support it by adding some Git config or something. But, we would need some evidence that users actually care about that.

Reply:
Sounds good.  I added some explanation to the FAQ, and also opened #441 to track any future enhancements.